File: CodinGame/Easy/Chuck Norris/ChuckNorris.cpp
#include <iostream>
#include <string>
#include <bitset>
#include <algorithm>

using namespace std;

int main()
{
    string MESSAGE;
    getline(cin, MESSAGE);
    int a = MESSAGE.length();

    bitset<700> Message;

    int Counter = 0;
    for (char &letter : MESSAGE)
    {
        bitset<7> Letter(letter);

        for (int i = 0; i < 7; i++)
            Message[Counter++] = Letter[6 - (Counter % 7)];
    }

    bool Current = !Message[0];
    bool Initial = true;

    for (int i = 0; i < Counter; i++)
    {
        if (Message[i] != Current)
        {
            if (!Initial)
                cout << " ";
            else
                Initial = false;

            if (Message[i])
                cout << "0";
            else
                cout << "00";

            cout << " ";
            Current = Message[i];
        }

        cout << "0";
    }
}

File: CodinGame/Easy/Chuck Norris/ChuckNorris.cs
using System;
using System.Linq;
using System.IO;
using System.Text;
using System.Collections;
using System.Collections.Generic;

namespace ChuckNorris
{
    static class Program
    {
        static void Main(string[] args)
        {

            string MESSAGE = Console.ReadLine();

            List<bool> fullBitArray = new List<bool>();

            foreach (char letter in MESSAGE)
            {
                BitArray reverseBitArray = new BitArray(BitConverter.GetBytes(letter));

                List<bool> bitArray = new List<bool>();

                for (int i = 0; i < 7; i++)
                    bitArray.Add(reverseBitArray[i]);

                bitArray.Reverse();

                fullBitArray.AddRange(bitArray);
            }

            bool current = new bool();
            bool initial = true;

            foreach (bool bit in fullBitArray)
            {
                if (bit != current || initial)
                {
                    current = bit;

                    if (!initial)
                        Console.Write(" ");
                    else
                        initial = false;

                    Console.Write(bit ? "0" : "00");
                    Console.Write(" ");
                }

                Console.Write("0");
            }

        }
    }
}